pylint/checkers/base.py

```diff
@@ -1693,28 +1707,36 @@ def _create_naming_options():
     name_options = []
     for name_type in sorted(KNOWN_NAME_TYPES):
```

Comment:
That would lead to quite a bit of code duplication though. I have added some line breaks and comments, perhaps that helps? If not, I'll change it. Let me know!